pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -95,38 +102,74 @@ def __init__(self, time, pqs):
         if self._pqs.shape[1] != 7:
             raise ValueError("`pqs` matrix shall have 7 columns.")
 
+        self._min_time = min(self.time)
+        self._max_time = max(self.time)
+
     def as_matrix(self, query_time):
+        """Get transformation matrix at given time.
+
+        Parameters
+        ----------
+        query_time : Union[float, array-like shape (...)]
+            Query time
+
+        Returns
+        -------
+        A2B_t : array, shape (4, 4) or (..., 4, 4)
+            Homogeneous transformation matrix / matrices at given time / times.
+        """
         pq = self._interpolate_pq_using_sclerp(query_time)
-        return transform_from_pq(pq)
+        transforms = transforms_from_pqs(pq)
+        transforms = np.squeeze(transforms)  # to keep the external API
+        return transforms
 
     def check_transforms(self):
         self._pqs[:, 3:] = norm_vectors(self._pqs[:, 3:])
         return self
 
     def _interpolate_pq_using_sclerp(self, query_time):
-        # identify the index of the preceding sample
-        idx_timestep_earlier_wrt_query_time = np.argmax(
-            self.time >= query_time) - 1
+        query_time_arr = np.atleast_1d(query_time)
 
-        # deal with first timestamp
-        idx_timestep_earlier_wrt_query_time = max(
-            idx_timestep_earlier_wrt_query_time, 0)
+        # identify the index of the preceding sample
```

Comment:
The handling of edge cases is way more complicated now. I think it is better now, but I'd like to hear your opinion (and need a review because the logic is considerably more complicated) @JonasHablitzel  @kopytjuk .

When the time is beyond the limits of the time array of this TimeVaryingTransform, I clip it to [min, max] and do not interpolate, I just return the first or last element. (Maybe the logic can be simplified a bit still.) Do you think the behavior makes sense. I am not entirely sure how these cases were handled before. I think an interpolation between the two first or last elements was done. Is that correct?

Reply:
I like the more complicated approach :) 
jup at least in my implementation this was  the case.

I also thought of to maybe expose a configuration flag in the `TimeVaryingTransform` if a user wants to have "extrapolation" or not. Because i think this can differ from usecase to usecase.

the flag could be:
```python
allow_extrapolation:bool=True
```

- if allow_extrapolation is enabled the logic would be the same as @AlexanderFabisch  suggested
- if its disabeld we would raise a ValueError if a tranform with a time larger or smaller then the [min,max] time is requested

what is your opinion on that? @AlexanderFabisch @kopytjuk

Reply:
For now (to finalize this PR) I would simply raise a ValueError if any of the elements of`query_time` array is outside of the provided transform samples. This would allow us to think about a nice control flag (e.g. as kwarg) in one of the next PRs. With that we can keep the existing logic and rewrite later if we want.

In any case I would avoid any automatisms, like automatic clipping. In my experience people just pass things, it does not crash and they debug for a very long time in their application code why something is wrong ... after hours/days they recognize that their timestamp might be the issue. Often two transforms, be it pq or homogenous matrices are hard to distinguish if they are different ...